tests/src/core/testqgsexpression.cpp

```diff
@@ -2875,6 +2875,15 @@ class TestQgsExpression: public QObject
       QTest::newRow( "translate" ) << "translate( $geometry, -1, 2)" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "LINESTRING (-1 2, 9 12)" ) );
       geom = QgsGeometry::fromPointXY( point );
       QTest::newRow( "translate" ) << "translate( $geometry, 1, -2)" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "POINT(1 -2)" ) );
+
+      geom = QgsGeometry::fromPolygonXY( polygon );
```

Comment:
I'd suggest adapting these tests to use geom_from_wkt instead of $geometry -- it means the tests will be stand alone and can be posted directly into the qgis app to help diagnose the results

Reply:
That is nice, and actually the way I tested the expressions manually. So it would be 

> +1 to centroid of bounding box.

Will try to make it optional. The centroid of the bbox is the same as the centroid of the geometry, right?